### ISSUE TITLE
Autocomplete is case sensitive

### ISSUE THREAD

**Issue by viharm** [firefly-iii/firefly-iii#2969]
I am running Firefly III version 4.8.2

**Description**

Currently auto-complete functionality in various text fields in the UI is case sensitive for instances with a _PostgreSQL_ database.

The desired behaviour is for this to be case-insensitive.

However when creating a recurring transaction, in the optional transaction information, the categories field allows auto-complete which is case insensitive. See screenshot below.

**Extra info**

This has been identified as a result of using _PostgreSQL_ in #2386. However if this functionality is case-insensitive for recurring transactions then there may be a way to achieve this elsewhere too.

**Bonus points**

![Case insensitive auto-complete example](https://user-images.githubusercontent.com/2663937/71783838-6f9f9600-2fe4-11ea-98d3-5e239fb69267.png)

**Comment by JC5** [comment]
If your database is case-sensitive, the data delivered to the auto-complete will be too.

Once cached however, the JS library handling the auto-complete may search its cache in an case insensitive fashion.

But the root cause is Postgres being case sensitive and that's a configuration thing, not a Firefly III thing.

**Comment by viharm** [comment]
@JC5, thanks for the information. However does this mean that Firefly III uses the JS cache for autocomplete for the recurring transactions and connects to the DB for other autocomplete functionality?

Just wondering if its possible in one place then why not others?

**Comment by JC5** [comment]
I'm not sure what to tell you except what I told you already.

**Comment by viharm** [comment]
@JC5, based on what you have said, it appears that Firefly III uses the cache for autocomplete for the recurring transactions (example in screenshot). Can it use the same cached data in other places too? Won't this make auto-complete case-insensitive - independent of the DB?

**Comment by JC5** [comment]
OK let me try again. The search in the database, for Postgres, is case sensitive. So if you search for `Flo` you will get Florian but not flower. Right?

The result is cached by your web browser. And then it becomes case-insensitive, apparantly. So then, `flo` results in both Florian and flower.

I can't cache the entire database in the browser now can I?

**Comment by viharm** [comment]
OK, understood; and thanks for your patience with me. However even if I try to repeat the same work flow in other areas, the cached lists are not used. Repeatedly, this is the only place where search is case-insensitive. The following screenshot is from from the same instance of Firefly III but whilst creating a new withdrawal.
![image](https://user-images.githubusercontent.com/2663937/71907852-ce8c1900-3164-11ea-87e3-10aef431bae1.png)

I now understand the caching logic and how the browser accesses the cache, but this behaviour is different for the two areas.

Is there any other way I can help provide more information?

Is anyone else having this issue?

**Comment by AlexKalopsia** [comment]
Just want to mention that I _love_ the fact that I can start typing something that is perhaps mid-sentence. So "My Personal Account (Savings)" can be quickly found by typing "sav". This behavior seems to be consistent across all my text fields in that form.

**Comment by viharm** [comment]
> 
> 
> Just want to mention that I _love_ the fact that I can start typing something that is perhaps mid-sentence. So "My Personal Account (Savings)" can be quickly found by typing "sav". This behavior seems to be consistent across all my text fields in that form.

@AlexKalopsia , are you using _PostgreSQL_ as your database?

**Comment by AlexKalopsia** [comment]
@viharm MariaDB

**Comment by viharm** [comment]
> 
> 
> @viharm MariaDB

@AlexKalopsia , this is expected. As per #2386 , _PostgreSQL_ causes it to be case-insensitive. Unfortunately we are not comparing like-for-like.
